services/bitbucket/bitbucket-pull-request.service.js

```diff
@@ -24,15 +25,57 @@ function pullRequestClassGenerator(raw) {
       })
     }
 
+    async fetchServer({ proto, hostAndPath, user, repo }) {
+      const url = `${proto}://${hostAndPath}/rest/api/1.0/projects/${user}/repos/${repo}/pull-requests`
+      const options = {
+        qs: {
+          state: 'OPEN',
+          limit: 100,
+          withProperties: false,
+          withAttributes: false,
+        },
+      }
+
+      if (
```

Comment:
The one _potential_ limitation I see with only including auth when hitting a BBS instance is that it means it won't be possible to access private BB repos via a self-hosted shields instance (the second scenario from a previous comment). 

I know your primary motivation here is to get support for BBS, but do we know if the same basic auth can work with BB (or can we determine quickly if so)?

Reply:
I think I only have a bitbucket cloud account through github, which I don't think will work with basic auth. I'll create an account to test with.

Reply:
Much appreciated! Don't spend too much time on this though. No one's asked about private BB repos so far and I don't want to hold up BBS support for that. If this is something we can cover quickly great but if not then we can roll with this as-is and cross the private BB repo bridge if/when it comes up

Reply:
Actually, now that I think about it more, I don't think this will work. If we want to support private repos on cloud we will have to add a separate secret, otherwise we won't be able to mix cloud and server requests on one instance, as the credentials for the BBS account probably don't work for BB. Thoughts?

Reply:
That's correct. We'd have two sets of creds, one for BB and another for BBS. I was thinking that could be handled by creating the `options` object in the main `fetch` method, doing the auth checks there, and passing the options object along to the respective fetch functions, or just adding a similar auth check to the `fetchCloud` function